### PR TITLE
Add option to preserve region names to the IntervalList dumper.

### PR DESCRIPTION
**Pull request by tmooney**
This will come in handy for the per-target metric reporting!

### CODE REVIEW

File: lib/perl/Genome/FeatureList/Command/DumpIntervalList.pm [hunk 1]
```diff
@@ -38,6 +43,10 @@ class Genome::FeatureList::Command::DumpIntervalList {
 sub execute {
     my $self = shift;
 
+    if ($self->preserve_region_names and $self->merge) {
+        $self->fatal_message('The --preserve-region-names option has no effect unless --nomerge is also specified.  Running the region merger reassigns the region names.');
```
Comment (jasonwalker80) [inline lib/perl/Genome/FeatureList/Command/DumpIntervalList.pm]: Is this true? When I have run the BED dump-interval-list command I often used `--merge` and `--noshort-name`.

Comment (tmooney) [reply lib/perl/Genome/FeatureList/Command/DumpIntervalList.pm]: I was surprised by this, too, but the testcase failed when I flipped `merge => 0` to `merge => 1`.  The long names were not preserved.

Comment (tmooney) [reply lib/perl/Genome/FeatureList/Command/DumpIntervalList.pm]: Looks like the trouble I was seeing is due to a long-standing issues with FeatureLists.  Asking to dump the merged list with different options does not actually do the work a second time--it returns the already-merged list from the first call.

At any rate, I've removed this error message from this command!

**Comment by tmooney** [comment]
CI, test this.

File: lib/perl/Genome/FeatureList/Command/DumpIntervalList.pm [hunk 2]
```diff
@@ -38,6 +43,9 @@ class Genome::FeatureList::Command::DumpIntervalList {
 sub execute {
     my $self = shift;
 
+#    if ($self->preserve_region_names and $self->merge) {
```
Comment (acoffman) [inline lib/perl/Genome/FeatureList/Command/DumpIntervalList.pm]: If this is going to be commented out, it should probably just be deleted?

Comment (tmooney) [reply lib/perl/Genome/FeatureList/Command/DumpIntervalList.pm]: Oops. It totally should!

**Pull request merged**